src/metabase/query_processor/middleware/update_used_cards.clj

```diff
@@ -12,14 +13,24 @@
 (set! *warn-on-reflection* true)
 
 (defn- update-used-cards!*
-  [used-card-ids]
-  (when (seq used-card-ids)
-    (qp.util/with-execute-async
-      (fn []
-        (try
-          (t2/update! :model/Card :id [:in used-card-ids] {:last_used_at :%now})
-          (catch Throwable e
-            (log/error e "Error updating used cards")))))))
+  [seq-of-used-card-ids]
+  (let [card-ids (apply set/union seq-of-used-card-ids)]
+    (log/debugf "Update last_used_at of %d cards" (count card-ids))
+    (try
+      ;; instead of updating each card with the timestmap that it was submited from the QP
+      ;; we update all cards with last_used_at = now to save db calls
+      ;; This value is not required to be accurate
+      (t2/update! :model/Card :id [:in card-ids] {:last_used_at :%now})
```

Comment:
want to call this out @luizarakaki, this means that all last_used_at will be set to the "batch processing time", not the exact time that it was recorded.

Currently, we process every 5 minutes, so last_used_at could be wrong up to 5 minutes.

This is a trade off between having to do hundreds of db calls vs accuracy.

Reply:
Is this a fundamental tradeoff? Could we not keep the timestamp around of when it was added to the queue?

Reply:
hmm, I guess I could use a big `case` here

Reply:
updated here: https://github.com/metabase/metabase/pull/45007/commits/d427b817b6d75666428c30556198c099dff69c12